packages/react-components/react-tree/src/components/TreeItemLayout/useTreeItemLayout.tsx

```diff
@@ -60,14 +116,39 @@ export const useTreeItemLayout_unstable = (
   const arrowNavigationProps = useArrowNavigationGroup({ circular: true, axis: 'horizontal' });
   const actions = isActionsVisible
     ? slot.optional(actionsShorthand, {
-        defaultProps: { ...arrowNavigationProps, role: 'toolbar' } as ExtractSlotProps<TreeItemLayoutSlots['actions']>,
+        defaultProps: { ...arrowNavigationProps, role: 'toolbar' },
         elementType: 'div',
       })
     : undefined;
-  const actionsRefs = useMergedRefs(actions?.ref, actionsRef);
+  const actionsRefs = useMergedRefs(actions?.ref, actionsRef, actionsRefInternal);
+  const handleActionsBlur = useEventCallback((event: React.FocusEvent<HTMLDivElement>) => {
+    if (isResolvedShorthand(actionsShorthand)) {
+      actionsShorthand.onBlur?.(event);
+    }
+    setIsActionsVisibleInternal(elementContains(event.currentTarget, event.relatedTarget as Node));
```

Comment:
```suggestion
    const IS_INSIDE_ACTIONS = elementContains(event.currentTarget, event.relatedTarget as Node)
    setIsActionsVisibleInternal(IS_INSIDE_ACTIONS);
```

You have been descriptive in other callbacks, let's follow the practice there (`IS_INSIDE_ACTIONS` - is just a template, please name it properly)